conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/DeadlineTagTranslator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.google.common.base.Strings;
+import com.palantir.deadlines.DeadlinesHttpHeaders;
+import com.palantir.tracing.TagTranslator;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.HeaderMap;
+import io.undertow.util.HttpString;
+
+/** A {@link TagTranslator} which populates metadata from deadlines headers. */
+enum DeadlineTagTranslator implements TagTranslator<HttpServerExchange> {
+    INSTANCE;
+
+    private static final HttpString EXPECT_WITHIN = HttpString.tryFromString(DeadlinesHttpHeaders.EXPECT_WITHIN);
```

Comment:
Sorry, missed this the first time.

`HttpString.tryFromString` can return `null` [(docs)](https://undertow.io/javadoc/1.4.x/io/undertow/util/HttpString.html#tryFromString-java.lang.String-).

I don't expect this, but just in case something changes in `undertow`, thoughts on adding a `Preconditions.checkNotNull(...)`?

Reply:
happy to change it, though we do not have equivalent protection in other places, e.g. in [CompletedRequestTagTranslator](https://github.com/palantir/conjure-java/blob/develop/conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/CompletedRequestTagTranslator.java#L32)

`HeaderMap#getFirst` will return null if the input is null (https://github.com/undertow-io/undertow/blob/main/core/src/main/java/io/undertow/util/HeaderMap.java#L47-L49)

Reply:
gotcha. fine to leave as-is then.

Reply:
@mpritham thoughts on this instead? [20fa9fa](https://github.com/palantir/conjure-java/pull/2529/commits/20fa9fa57c3505680a7197f11effdea9f04dce4f)

I'd prefer this code not to throw via `checkNotNull`, if for some reason `tryFromString` returns null, we should just omit the tags.

Reply:
Looks good!